### ISSUE TITLE
Coverage fails on documentation using scribble/srcdoc

### ISSUE THREAD

**Issue by lexi-lambda** [florence/cover#79]
This is probably another submodule bug, not a problem with cover specifically, but I figured I'd open an issue just to document the problem.

It would seem that documentation created with `scribble/srcdoc` creates a submodule within the containing module, and using `include-extracted` pulls information from that submodule. When cover runs the documentation module, I get an error:

```
do-lift.57: undefined;
 cannot reference an identifier before its definition
  in module: (submod "/Users/alexis/gits/alexis-collection-lens/collection/lens.rkt" srcdoc)
  context...:
   /Users/alexis/gits/scribble/scribble-lib/scribble/private/manual-proc.rkt:212:0: *defproc14
   (submod /Users/alexis/gits/alexis-collection-lens/collection/lens.rkt srcdoc): [running body]
   /Users/alexis/gits/alexis-collection-lens/scribblings/main.scrbl: [traversing imports]
   /Users/alexis/gits/racket/racket/share/pkgs/cover/cover.rkt:88:0: handle-file
   /Users/alexis/gits/racket/racket/share/pkgs/cover/cover.rkt:72:8: for-loop
   /Users/alexis/gits/racket/racket/share/pkgs/cover/cover.rkt:53:0
   (submod /Users/alexis/gits/racket/racket/share/pkgs/cover/raco.rkt main): [running body]
   /Users/alexis/gits/racket/racket/collects/raco/raco.rkt: [running body]
   /Users/alexis/gits/racket/racket/collects/raco/main.rkt: [running body]
```

The coverage report still finishes, but since the documentation errors, it is not included in the coverage report, and the pieces of documentation in the source module are considered uncovered, which lowers the test coverage amount substantially.

**Comment by florence** [comment]
This is exactly the error that occurs when cover interacts poorly with submodule.

A potential fix is to make sure that the file with the problematic submodule is compiled *first* (ie put it first in cover's argument list). That sometimes fixes the problem....

You could also exclude this file from the report with `-n`, exclude all `.scbl` files with `-b` (`-b` excludes other things, like tests too), or skip covering the file entirely with `test-omit-paths` or `cover-omit-paths` in your `info.rkt`.

**Comment by lexi-lambda** [comment]
In this case, [more or less all of my code is in a module using `scribble/srcdoc`](https://github.com/lexi-lambda/alexis-collection-lens/blob/master/collection/lens.rkt), so ignoring the file sort of defeats the point. That said, it clearly can't really be helped—hopefully the new macro expander will solve the problem.

**Comment by florence** [comment]
Would it help if there were a flag to tell cover to skip annotating some submodules?

Something like `--skip-submodule include-extracted`. Cover already (by default) excludes submodules from its coverage report, it doesn't make sense to annotate those submodules. (Perhaps the already existing `-e` flag could do this).

It would take a while to make this change, as it would require writing a new annotator (errortrace can't do this), but thats already on my TODO list.

**Comment by florence** [comment]
Fixed by #81
